### PR TITLE
feat: create use memberAction hook

### LINKED ISSUE #745: Create a new hook to get user actions general insights

### PR DESCRIPTION
**Pull request by LinaYahya**
closes #745

### CODE REVIEW

File: src/config/keys.ts [hunk 1]
```diff
@@ -282,6 +282,17 @@ export const buildActionsKey = (args: {
   },
 ];
 
+export const buildMemberActionsFilteredByDateKey = (args: {
```
Comment (pyphilia) [inline src/config/keys.ts:285]:
```suggestion
export const buildMemberActionsKey = (args: {
```
I think on longer term we will add more filters, so no need to be too precise.

**Review by pyphilia — APPROVED** [review]
LGTM! 🖍️

File: src/api/action.ts [hunk 2]
```diff
@@ -17,6 +18,14 @@ export const getActions = async (
     .get<ActionData>(`${API_HOST}/${buildGetActions(args.itemId, args)}`)
     .then(({ data }) => data);
 
+export const getMemberActionsFilteredByDate = async (
```
Comment (spaenleh) [inline src/api/action.ts:21]:
```suggestion
export const getMemberActions = async (
```

File: src/api/routes.ts [hunk 3]
```diff
@@ -353,6 +353,21 @@ export const buildGetActions = (
       addQueryPrefix: true,
     },
   )}`;
+
+export const buildGetMemberActions = (args: {
```
Comment (spaenleh) [inline src/api/routes.ts:357]:
```suggestion
export const buildGetMemberActionsRoute = (args: {
```

**Review by spaenleh — CHANGES_REQUESTED** [review]

File: src/config/keys.ts (hunk 1 above)
Comment (spaenleh) [inline src/config/keys.ts:285]: Should this key be part of the `memberKeys.current()` ?

File: src/hooks/action.test.ts [hunk 4]
```diff
@@ -166,4 +174,46 @@ describe('Action Hooks', () => {
       expect(queryClient.getQueryData(key)).toBeFalsy();
     });
   });
+
+  describe('useMemberActions', () => {
+    const args = {
+      startDate: '2024-04-24T14:07:00.074Z',
+      endDate: '2024-04-24T14:07:00.074Z',
```
Comment (spaenleh) [inline src/hooks/action.test.ts:181]: Start and end date are the same, is this intentional ?

File: src/hooks/action.test.ts [hunk 5]
```diff
@@ -166,4 +174,46 @@ describe('Action Hooks', () => {
...
+    };
+    const route = `/${buildGetMemberActionsRoute(args)}`;
+    const key = memberKeys.current().actions(args);
+
+    const response = ACTIONS_DATA;
+
+    it(`Receive actions for current member`, async () => {
+      const hook = () => hooks.useMemberActions(args);
```
Comment (spaenleh) [inline src/hooks/action.test.ts:189]: Do you need to redefine the hook in each test ? can it be defined in the describe instead ?

**Review by spaenleh — APPROVED** [review]
Changes look good to me, just some small nitpicks. You can merge once fixed !

**Pull request merged**